### ISSUE TITLE
isProbablyReadable false positive when page uses <br> instead of <p>

### ISSUE THREAD

**Issue by salty-horse** [mozilla/readability#247]
The websites Livejournal and its fork, Dreamwidth, seem to publish blog posts using `<br>` tags for paragraphs instead of `<p>`.

[Example page on dreamwidth.org](https://ladybusiness.dreamwidth.org/2015/08/02/station-eleven-emily-st-john-mandel.html)

As a consequence, pages that Readability can parse() easily, fail on the isProbablyReaderable() test, which means that Firefox doesn't show the Reader View button.

I have raised the issue to @gijsk on IRC. Here is the chat log, posted by request.

```
<Gijs> salty-horse: if instead of all the <br> the text in the content used <p> tags, it'd work. :)
<Gijs> salty-horse: and yeah, sadly github didn't let me have "gijs"
<Gijs> (I was late to the github party, what can I say...)
<Gijs> salty-horse: isProbablyReaderable is very stupid
<Gijs> somewhat intentionally
<Gijs> it runs on every page load
<Gijs> so for hopefully obvious reasons we want it to be very very fast :)
<Gijs> making it smarter will mean making it slower
<salty-horse> so, what's the recommended course of action? :)
<Gijs> if you control the markup of the page, fixing the markup seems like the easiest solution
<salty-horse> thinking outside of Readability, why not allow FF to "force" reader view?
<Gijs> if you just want to use readability more often, it would be pretty trivial to create an add-on that forced that button to always be visible
<salty-horse> Gijs, I have a feeling it's an issue with the Livejournal authoring tools, not the specific page
<Gijs> yes, that's plausible
<Gijs> that or the markup was copied from msword
<Gijs> or... well, there are a number of other options
<Gijs> either way, if you don't control the markup, you could try to force the button to be visible more often
<Gijs> as for why we don't do that
<Gijs> it's an 80/20 type thing
<Gijs> the failure behaviour for the button not working is pretty shitty UX
<Gijs> it used to be "reload the original page"
<Gijs> IIRC we changed it now to just show you a message along the lines of "uh oh, it seems that didn't work"
<Gijs> because the lack of feedback was concerning
<Gijs> ie, there are obviously pages where readability just goes "uhm, I can't make this readerable"
<Gijs> either correctly or incorrectly
<salty-horse> and there's no will to change isProbablyReaderable to allow <br>'s because it's either too slow or creates too many false positives?
<Gijs> hm
<Gijs> it would depend on how slow, exactly
<Gijs> clearly we can't *just* count the number of <br>s
<salty-horse> should I open a bug and have it discussed there?
<salty-horse> counting br's with text between them?
<Gijs> so right now we count up text in <p>s in isProbablyReaderable, IIRC
<Gijs> I think the ideal solution would add containers that have a lot of <br> kids
<Gijs> (to have their text counted, too)
<Gijs> but I don't know that I can think of a way of doing that would perform reasonably.
<Gijs> salty-horse: IIRC the nodejs-based test stuff in the readability repo has benchmarks
<Gijs> ... however, I think they run on the fake DOM stuff
<Gijs> salty-horse: I *think* you might be able to use something like document.querySelectorAll("br:nth-of-type(10)")
<Gijs> for some yet-to-be-determined-N instead of 10
<Gijs> to semi-efficiently get the type of <br> we're interested in
<Gijs> (and thereby its parent)
<Gijs> (or their parents, rather)
<Gijs> but I expect that won't work (well) on the fake DOM stuff
<Gijs> I *think* we run isProbablyReaderable on gecko though
<Gijs> in which case I'd be less scared
* Gijs pokes
<Gijs> yeah, we do
```

**Comment by salty-horse** [comment]
Seems fixed in Firefox 52.0.1

**Comment by gijsk** [comment]
Yes, this was fixed in #310 / https://bugzilla.mozilla.org/show_bug.cgi?id=1177619 which landed for 52. Thanks for updating the issue!
